amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/transformation/stage/ParametersNormalizationStage.scala

```diff
@@ -25,34 +26,58 @@ abstract class ParametersNormalizationStage(profile: ProfileName) extends Transf
                          configuration: AMFGraphConfiguration): BaseUnit = model match {
     case doc: Document if doc.encodes.isInstanceOf[Api] =>
       val api = doc.encodes.asInstanceOf[Api]
-      transform(api)
+      transform(api)(errorHandler)
       doc
     case _ => model
   }
 
-  protected def transform(api: Api): Api = api
+  protected def transform(api: Api)(implicit errorHandler: AMFErrorHandler): Api = api
 
-  protected def pushParamsToEndpointOperations(endpoint: EndPoint, finalParams: Parameters) = {
+  protected def pushParamsToEndpointOperations(endpoint: EndPoint, finalParams: Parameters)(
+      implicit errorHandler: AMFErrorHandler) = {
     endpoint.operations.foreach { op =>
-      setRequestParameters(op, finalParams)
+      setRequestParameters(op, finalParams)(errorHandler)
     }
   }
 
   protected def assignPathParametersTo(endpoint: EndPoint, path: Seq[Parameter]) =
     if (path.nonEmpty)
       endpoint.fields.setWithoutId(EndPointModel.Parameters, AmfArray(path))
 
-  private def setRequestParameters(op: Operation, params: Parameters) = {
-    val request = Option(op.request).getOrElse(op.withRequest())
-
-    val finalParams = params.merge(Parameters(request.queryParameters, request.uriParameters, request.headers))
+  private def setRequestParameters(op: Operation, params: Parameters)(implicit errorHandler: AMFErrorHandler) = {
+    val request           = Option(op.request).getOrElse(op.withRequest())
+    val requestParameters = Parameters(request.queryParameters, request.uriParameters, request.headers)
+    checkDuplicatedParameters(params, requestParameters, errorHandler)
+    val finalParams = params.merge(requestParameters)
     // set the list of parameters at the operation level in the corresponding fields
     if (finalParams.query.nonEmpty)
       request.fields.setWithoutId(RequestModel.QueryParameters, AmfArray(finalParams.query))
     if (finalParams.header.nonEmpty) request.fields.setWithoutId(RequestModel.Headers, AmfArray(finalParams.header))
     val pathParams = finalParams.baseUri08 ++ finalParams.path
     if (pathParams.nonEmpty) request.fields.setWithoutId(RequestModel.UriParameters, AmfArray(pathParams))
   }
+
+  private def checkDuplicatedParameters(endPointParams: Parameters,
+                                        requestParams: Parameters,
+                                        errorHandler: AMFErrorHandler): Unit = {
+    val duplicates = endPointParams.findDuplicatesIn(requestParams)
+    duplicates.foreach({ parameter =>
```

Comment:
```scala
duplicates.foreach {  parameter =>
        addWarningForDuplicateParameter(parameter)
    }
```
Omitted errorHandler because it can be implicit.